dot/node.go

```diff
@@ -442,7 +490,7 @@ func (n *Node) Stop() {
 	}
 }
 
-func loadRuntime(cfg *Config, ns *runtime.NodeStorage,
+func (nodeBuilder) loadRuntime(cfg *Config, ns *runtime.NodeStorage,
```

Comment:
nit I'd suggest to have pointer receivers since interfaces are pointers

```suggestion
func (n *nodeBuilder) loadRuntime(cfg *Config, ns *runtime.NodeStorage,
```